src/components/font_size_editor/font_size_editor.ts

```diff
@@ -0,0 +1,109 @@
+import { Component, useExternalListener, useRef, useState } from "@odoo/owl";
+import { DEFAULT_FONT_SIZE, FONT_SIZES } from "../../constants";
+import { clip } from "../../helpers/index";
+import { setStyle } from "../../registries/index";
+import { SpreadsheetChildEnv } from "../../types/index";
+import { css } from "../helpers/css";
+import { isChildEvent } from "../helpers/dom_helpers";
+
+interface State {
+  isOpen: boolean;
+}
+
+interface Props {
+  onDone: () => void;
+  dropdownStyle: string;
+}
+
+// -----------------------------------------------------------------------------
+// TopBar
+// -----------------------------------------------------------------------------
+css/* scss */ `
+  .o-font-size-editor {
+    input.o-font-size {
+      height: 20px;
+      width: 23px;
+    }
+    input[type="number"] {
+      -moz-appearance: textfield;
+    }
+    input::-webkit-outer-spin-button,
+    input::-webkit-inner-spin-button {
+      -webkit-appearance: none;
+    }
+  }
+`;
+
+export class FontSizeEditor extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-FontSizeEditor";
+  static components = {};
+  fontSizes = FONT_SIZES;
+
+  dropdown: State = useState({ isOpen: false });
+  openedEl: HTMLElement | null = null;
+
+  private inputRef = useRef("inputFontSize");
+  private rootEditorRef = useRef("FontSizeEditor");
+
+  setup() {
+    useExternalListener(window, "click", this.onExternalClick);
+  }
+
+  onExternalClick(ev: MouseEvent) {
+    if (!isChildEvent(this.rootEditorRef.el!, ev)) {
+      this.closeFontList();
+    }
+  }
+
+  get currentFontSize(): number {
+    return this.env.model.getters.getCurrentStyle().fontSize || DEFAULT_FONT_SIZE;
+  }
+
+  toggleFontList() {
+    const isOpen = this.dropdown.isOpen;
+    if (!isOpen) {
+      this.props.onDone();
```

Comment:
`onDone` is not very clear. I do not understand well what is the use of that function

Reply:
Yes, indeed. Changed to onToggled (it contains the method from the topbar that should be called when toggling the list) and removed the call as 2 useless places